hazelcast/src/main/java/com/hazelcast/query/Predicates.java

```diff
@@ -514,12 +514,16 @@ public static PredicateBuilder newPredicateBuilder() {
      * Creates a paging predicate with an inner predicate and page size. Results will be filtered via inner predicate
      * and will be returned in natural order.
      *
+     * <b>Object being compared must implement {@link java.lang.Comparable}.  
+     * Use custom comparator if Object does not implement {@link java.lang.Comparable}</b>
```

Comment:
Can you clarify if the entry key or the value needs to be `Comparable`? And avoid using `Object` (capital case) as I assume then you wanted to refer to `java.lang.Object` instead of some generic object.
Also, can you mention here that the query execution will fail with the `IllegalArgumentException` instead of mentioning it in `@throws` clause? Since this method will never throw that exception.